### ISSUE TITLE
Steps misbehaved when I call capture() function onDie.

### ISSUE THREAD

**Issue by vimagick** [casperjs/casperjs#1317]
I'm running phantomjs (2.0.0) + casperjs (latest) on Ubuntu/MacOSX.

```
$ cat die.js

var casper = require('casper').create({
  logLevel: 'debug',
  verbose: true,
});

casper.options.onDie = function(_, message, status){
  this.capture('die.png');
}

casper.start('about:blank');

casper.then(function() {
        this.die('xxx');
      })
      .then(function() {
        this.die('yyy');
      })
      .then(function() {
        this.die('zzz');
      });

casper.run();
```

```
$ casperjs die.js

[info] [phantom] Starting...
[info] [phantom] Running suite: 5 steps
[debug] [phantom] opening url: about:blank, HTTP GET
[debug] [phantom] url changed to ""
[debug] [phantom] Successfully injected Casper client-side utilities
[debug] [phantom] start page is loaded
[error] [phantom] xxx
xxx
[debug] [phantom] Capturing page to /private/tmp/die.png
[info] [phantom] Capture saved to /private/tmp/die.png
[info] [phantom] Step anonymous 3/5: done in 152ms.
[error] [phantom] yyy
yyy
[debug] [phantom] Capturing page to /private/tmp/die.png
[info] [phantom] Capture saved to /private/tmp/die.png
[info] [phantom] Step anonymous 4/5: done in 189ms.
```

If I comment `this.capture('die.png')` out.

```
$ sed -i '/capture/s@^@//@' die.js

$ casperjs die.js

[info] [phantom] Starting...
[info] [phantom] Running suite: 5 steps
[debug] [phantom] opening url: about:blank, HTTP GET
[debug] [phantom] url changed to ""
[debug] [phantom] Successfully injected Casper client-side utilities
[debug] [phantom] start page is loaded
[error] [phantom] xxx
xxx
[info] [phantom] Step anonymous 3/5: done in 72ms.
```

**Comment by BIGjuevos** [comment]
**As part of a cleanup effort:**
Looks to be a stale help request. Assuming you've moved on to better and cooler things. Please feel free to re-open if this is still an active issue for you and we'll try our best to help out.

**Comment by vimagick** [comment]
This issue still exists. Can I reopen it?

**Comment by jefleponot** [comment]
Hi @vimagick , thanks for your contribution
Thank you especially for persevering.

I test you example and I have the same result on phantomjs 2.1.1
it works with slimerJS 0.10.2

```js 
[info] [phantom] Starting... 
[info] [phantom] Running suite: 5 steps 
[debug] [phantom] opening url: about:blank, HTTP GET 
[debug] [phantom] url changed to "about:blank" 
[debug] [phantom] Successfully injected Casper client-side utilities 
[debug] [phantom] start page is loaded 
[error] [phantom] xxx 
xxx 
[debug] [phantom] Capturing page to /home/jef/dev/casperjs/die.png 
[info] [phantom] Capture saved to /home/jef/dev/casperjs/die.png 
[info] [phantom] Step anonymous 3/5: done in 486ms.
```

CasperJS use an asynchronous call to stop the program execution. So multiple calls of the "die" method are authorised.
Here is a proposal Fix : #1726

**Comment by kensoh** [comment]
@vimagick happy to update that Jef's PR had been merged and your script seems to work correctly now using CasperJS v1.1.4 and PhantomJS v2.1.1. Closing the issue but do ping if anything :)

```
[info] [phantom] Starting...
[info] [phantom] Running suite: 5 steps
[debug] [phantom] opening url: about:blank, HTTP GET
[debug] [phantom] url changed to ""
[debug] [phantom] Successfully injected Casper client-side utilities
[debug] [phantom] start page is loaded
[error] [phantom] xxx
xxx
[debug] [phantom] Capturing page to /Users/kensoh/Cloud Drive/Marketing/Website/api/tagui/flows/die.png
[info] [phantom] Capture saved to /Users/kensoh/Cloud Drive/Marketing/Website/api/tagui/flows/die.png
[info] [phantom] Step anonymous 3/5: done in 161ms.
[info] [phantom] Step anonymous 4/5: done in 162ms.
[debug] [phantom] Navigation requested: url=about:blank, type=Other, willNavigate=true, isMainFrame=true
[debug] [phantom] url changed to "about:blank"
```
